Source/WebGPU/WebGPU/BindGroupLayout.mm

```diff
@@ -224,12 +256,23 @@ static bool isPresent(const WGPUStorageTextureBindingLayout& storageTexture)
 {
 }
 
+BindGroupLayout::BindGroupLayout(id<MTLBuffer> vertexArgumentBuffer, id<MTLBuffer> fragmentArgumentBuffer, id<MTLBuffer> computeArgumentBuffer, HashMap<uint32_t, uint32_t>&& shaderStageForBinding, Device& device)
+    : m_vertexArgumentBuffer(vertexArgumentBuffer)
+    , m_fragmentArgumentBuffer(fragmentArgumentBuffer)
+    , m_computeArgumentBuffer(computeArgumentBuffer)
+    , m_device(device)
+    , m_shaderStageForBinding(WTFMove(shaderStageForBinding))
+{
+}
+
 BindGroupLayout::BindGroupLayout(Device& device)
     : m_device(device)
 {
 }
 
-BindGroupLayout::~BindGroupLayout() = default;
+BindGroupLayout::~BindGroupLayout()
```

Comment:
Any reason for this?

Reply:
Yes, I added a union with an objective-c type, so defaulting the destructor results in:
```
/Volumes/Data/OpenSource/Source/WebGPU/WebGPU/BindGroupLayout.mm:273:18: error: defaulting this destructor would delete it after its first declaration
BindGroupLayout::~BindGroupLayout() = default;
In file included from /Volumes/Data/OpenSource/Source/WebGPU/WebGPU/BindGroupLayout.mm:27:
/Volumes/Data/OpenSource/Source/WebGPU/WebGPU/BindGroupLayout.h:83:38: note: destructor of '' is implicitly deleted because variant field 'm_vertexArgumentEncoder' is an ObjC pointer
        const id<MTLArgumentEncoder> m_vertexArgumentEncoder { nil };
```

However I think I can avoid using the union which would allow me to revert this change, I will take a look

Reply:
I tried and that change is a lot larger than non-defaulting the destructor, it requires implementing a common protocol which is fine but then also casting up to the appropriate type which the union handles implicitly. I think I will leave this as-is if that is ok